config/configgrpc/bearer_token.go

```diff
@@ -15,44 +15,45 @@
 package configgrpc
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"io/ioutil"
-	"regexp"
 
 	"google.golang.org/grpc/credentials"
 )
 
 var _ credentials.PerRPCCredentials = (*PerRPCAuth)(nil)
 
-// PerRPCAuth is a gRPC credentials.PerRPCCredentials implementation that returns an 'authorization' header
+// PerRPCAuth is a gRPC credentials.PerRPCCredentials implementation that returns an 'authorization' header.
 type PerRPCAuth struct {
 	metadata map[string]string
 }
 
-// BearerTokenFromFile builds a new PerRPCAuth with bearer token authentication, reading the token from the specified file
+// BearerTokenFromFile builds a new PerRPCAuth with bearer token authentication, reading the token from the specified file.
 func BearerTokenFromFile(file string) (*PerRPCAuth, error) {
 	token, err := ioutil.ReadFile(file)
 	if err != nil {
 		return nil, fmt.Errorf("couldn't obtain token from file: %w", err)
 	}
 
-	re := regexp.MustCompile(`[\s\n]`)
-	token = re.ReplaceAll(token, []byte(""))
+	// Replace all white space.
+	token = bytes.ReplaceAll(token, []byte(" "), []byte(""))
```

Comment:
Sanity check: when it was a regexp it was `\s` so it was the same was `[\t\n\f\r ]` but I assume that the intention is really to just remove single whitespace char + newline.

Reply:
Right: with the data I was using, the problem was really about the `\n`, but I decided to use ` \s` as well as it was cheap.